File: semester4/web-programming/angular-php/Server/Model/Ingredient.php
<?php
    class Ingredient {
        public $IngredientID;
        public $Name;
        public $Quantity;
        public $MeasurementUnit;
        public $RecipeID;

        public function isValid() {
            return !empty($this->Name) && 
                   $this->Quantity > 0 && 
                   !empty($this->MeasurementUnit) && 
                   !empty($this->RecipeID);
        }
    }
?> 

File: semester4/web-programming/assignments/asp-dotnet/Models/Ingredient.cs
namespace asp_dotnet.Models
{
    public class Ingredient
    {
        public int IngredientID { get; set; }
        public string? Name { get; set; }
        public int Quantity { get; set; }
        public string? MeasurementUnit { get; set; }
        public int RecipeId { get; set; }
    }
}